docs/source/notes/libtorch_stable_abi.md

```diff
@@ -0,0 +1,48 @@
+# LibTorch Stable ABI
+
+This note will eventually contain more details on how to use the APIs in torch/csrc/stable. For the moment, it contains a table of internal representations:
+1. type in custom extension: type used within the end user custom library.
+2. StableIValue representation: a stable conversion of the type to liaison between the user model vs libtorch.so.
+3. LibTorch type: type used within libtorch.so.
+4. Schema Type: type as described by the schema, which we hail as the source of truth for both ATen ops in native_functions.yaml and for user defined custom ops registered to the dispatcher.
+
+|  type in custom extension    |   StableIValue representation   |   type in libtorch  |   IValue  | 
+| -------- | ------- | ------- | ------- |
+| std::optional\<S> | \*reinterpret_cast\<(StableIValue\*)\*>, pointer to a StableIValue recursively defined | std::optional\<T> | OptionalType |
+| std::nullopt | \*reinterpret_cast\<nullptr_t\*> | IValue() | NoneType |
```

Comment:
not correct anymore.

Reply:
no this one is correct?

Reply:
Umm schemas don't exactly have NoneType in them

Reply:
I changed the table so it uses the schema types like in native_functions.yaml so things should be clearer now